File: 2608-count-the-digits-that-divide-a-number/count-the-digits-that-divide-a-number.js
/**
 * @param {number} num
 * @return {number}
 */
var countDigits = function(num) {
    let count = 0;
    let digits = getDigit(num);
    let result = digits.next();
    while (!result.done) {
        if (num % result.value === 0) count++;
        result = digits.next();
    }
    return count;
};

function *getDigit(num) {
    while (num > 0) {
        yield num % 10;
        num = Math.floor(num / 10);
    }
}

File: 2608-count-the-digits-that-divide-a-number/count-the-digits-that-divide-a-number.java
class Solution {
    public int countDigits(int num) {
        String val = num+"";
        int count =0;
        for(int i=0; i<val.length(); i++){

      
            if(num%(val.charAt(i)-'0')==0)count++;
        }
        return count;
    }
}